tools/ci/prepare_release_assets.py

```
@@ -0,0 +1,394 @@
+    dest.parent.mkdir(parents=True, exist_ok=True)
+
+    headers = {
+        "User-Agent": "StarRail-OneDragon CI",
+        "Accept": "application/octet-stream",
+    }
+    if token:
+        headers["Authorization"] = f"Bearer {token}"
+
+    for attempt in range(1, retries + 1):
+        try:
+            req = urllib.request.Request(url, headers=headers)
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)